boards/arm/stm32f746g_disco/early_init.c

```diff
@@ -0,0 +1,156 @@
+	register __IO u32_t index;
+
+	/* Reset the RCC clock configuration */
+	RCC->CR = (u32_t)0x00000083;
+	RCC->CFGR = (u32_t)0x00000000;
+	RCC->PLLCFGR = (u32_t)0x24003010;
+	RCC->CIR = (u32_t)0x00000000;
+
+	/* Enable GPIO clocks for FMC pins */
+	RCC->AHB1ENR |= (
+		RCC_AHB1ENR_GPIOCEN |
+		RCC_AHB1ENR_GPIODEN |
+		RCC_AHB1ENR_GPIOEEN |
+		RCC_AHB1ENR_GPIOFEN |
+		RCC_AHB1ENR_GPIOGEN |
+		RCC_AHB1ENR_GPIOHEN
+	);
+
+	/* Cofigure the GPIO Pins for FMC */
+	configure_gpio_fmc('C', 3);
+	configure_gpio_fmc('D', 0);
+	configure_gpio_fmc('D', 1);
+	configure_gpio_fmc('D', 8);
+	configure_gpio_fmc('D', 9);
+	configure_gpio_fmc('D', 10);
+	configure_gpio_fmc('D', 14);
+	configure_gpio_fmc('D', 15);
+	configure_gpio_fmc('E', 0);
+	configure_gpio_fmc('E', 1);
+	configure_gpio_fmc('E', 7);
+	configure_gpio_fmc('E', 8);
+	configure_gpio_fmc('E', 9);
+	configure_gpio_fmc('E', 10);
+	configure_gpio_fmc('E', 11);
+	configure_gpio_fmc('E', 12);
+	configure_gpio_fmc('E', 13);
+	configure_gpio_fmc('E', 14);
+	configure_gpio_fmc('E', 15);
+	configure_gpio_fmc('F', 0);
+	configure_gpio_fmc('F', 1);
+	configure_gpio_fmc('F', 2);
+	configure_gpio_fmc('F', 3);
+	configure_gpio_fmc('F', 4);
+	configure_gpio_fmc('F', 5);
+	configure_gpio_fmc('F', 11);
+	configure_gpio_fmc('F', 12);
+	configure_gpio_fmc('F', 13);
+	configure_gpio_fmc('F', 14);
+	configure_gpio_fmc('F', 15);
+	configure_gpio_fmc('G', 0);
+	configure_gpio_fmc('G', 1);
+	configure_gpio_fmc('G', 4);
+	configure_gpio_fmc('G', 5);
+	configure_gpio_fmc('G', 8);
+	configure_gpio_fmc('G', 15);
+	configure_gpio_fmc('H', 3);
+	configure_gpio_fmc('H', 5);
+
+	/* Enable the FMC interface clock */
+	RCC->AHB3ENR |= RCC_AHB3ENR_FMCEN;
+
+	/* Configure and enable SDRAM bank1 */
+	FMC_Bank5_6->SDCR[0] = 0x00001954;
+	FMC_Bank5_6->SDTR[0] = 0x01115351;
+
+	/* SDRAM initialization sequence */
+	/* Clock enable command */
+	FMC_Bank5_6->SDCMR = 0x00000011;
+	temp_register = FMC_Bank5_6->SDSR & 0x00000020;
+	while ((temp_register != 0) && (timeout-- > 0)) {
+		temp_register = FMC_Bank5_6->SDSR & 0x00000020;
+	}
+
+	/* Delay */
+	for (index = 0; index < 1000; index++) {
+	}
+
+	/* PALL command */
+	FMC_Bank5_6->SDCMR = 0x00000012;
+	timeout = 0xFFFF;
+	while ((temp_register != 0) && (timeout-- > 0)) {
+		temp_register = FMC_Bank5_6->SDSR & 0x00000020;
+	}
+
+	/* Auto refresh command */
+	FMC_Bank5_6->SDCMR = 0x000000F3;
+	timeout = 0xFFFF;
+	while ((temp_register != 0) && (timeout-- > 0)) {
+		temp_register = FMC_Bank5_6->SDSR & 0x00000020;
+	}
+
+	/* MRD register program */
+	FMC_Bank5_6->SDCMR = 0x00044014;
+	timeout = 0xFFFF;
+	while ((temp_register != 0) && (timeout-- > 0)) {
+		temp_register = FMC_Bank5_6->SDSR & 0x00000020;
+	}
+
+	/* Set refresh count */
+	temp_register = FMC_Bank5_6->SDRTR;
+	FMC_Bank5_6->SDRTR = (temp_register | (0x0000050C << 1));
+
+	/* Disable write protection */
+	temp_register = FMC_Bank5_6->SDCR[0];
+	FMC_Bank5_6->SDCR[0] = (temp_register & 0xFFFFFDFF);
```

Comment:
What about using HAL api (as it seems LL is not available) to make this code STM32 generic ?

Reply:
I can use HAL... :)